### PR TITLE
permission sync in cloud and beat expiry adjustment

### PR DESCRIPTION
**Pull request by rkuo-danswer**
## Description
[Provide a brief description of the changes in this PR]

## How Has This Been Tested?
[Describe the tests you ran to verify your changes]

## Backporting (check the box to trigger backport action)
Note: You have to check that the action passes, otherwise resolve the conflicts manually and tag the patches.
- [ ] This PR should be backported (make sure to check that the backport attempt succeeds)

### CODE REVIEW

File: backend/onyx/background/celery/tasks/indexing/tasks.py [hunk 1]
```diff
@@ -911,14 +937,16 @@ def connector_indexing_task_wrapper(
...
+    except Exception:
         logger.exception(
             f"connector_indexing_task exceptioned: "
             f"tenant={tenant_id} "
             f"index_attempt={index_attempt_id} "
             f"cc_pair={cc_pair_id} "
             f"search_settings={search_settings_id}"
         )
+
+        sys.exit(255)
```
Comment (pablonyx) [inline backend/onyx/background/celery/tasks/indexing/tasks.py:960]: Was this intentional? We'll never reach the exception raised below this line

File: backend/onyx/background/celery/tasks/indexing/tasks.py [hunk 2]
```diff
@@ -991,7 +1019,17 @@ def connector_indexing_task(
...
         )
 
+    # this wait is needed to avoid a race condition where
+    # the primary worker sends the task and it is immediately executed
+    # beore the primary worker can finalize the fence
+    start = time.monotonic()
     while True:
+        if time.monotonic() - start > 300:
+            raise ValueError(
+                f"connector_permission_sync_generator_task - timed out waiting for fence to be ready: "
```
Comment (pablonyx) [inline backend/onyx/background/celery/tasks/indexing/tasks.py]: Did we mean `connector_indexing_task`?

File: backend/onyx/background/celery/tasks/doc_permission_syncing/tasks.py [hunk 3]
```diff
@@ -219,6 +221,43 @@ def connector_permission_sync_generator_task(
 
     r = get_redis_client(tenant_id=tenant_id)
 
+    # this wait is needed to avoid a race condition where
+    # the primary worker sends the task and it is immediately executed
+    # beore the primary worker can finalize the fence
```
Comment (pablonyx) [inline backend/onyx/background/celery/tasks/doc_permission_syncing/tasks.py]: Nit: before

File: backend/onyx/background/celery/tasks/doc_permission_syncing/tasks.py [hunk 4]
```diff
@@ -219,6 +221,43 @@ def connector_permission_sync_generator_task(
...
+    start = time.monotonic()
+    while True:
+        if time.monotonic() - start > 300:
```
Comment (pablonyx) [inline backend/onyx/background/celery/tasks/doc_permission_syncing/tasks.py]: Don't feel very strongly about this, but consider using a constant in place of a magic number here if you haven't already

File: backend/onyx/background/celery/tasks/doc_permission_syncing/tasks.py [hunk 5]
```diff
@@ -219,6 +221,43 @@ def connector_permission_sync_generator_task(
...
+            raise ValueError(
+                "connector_permission_sync_generator_task: payload invalid or not found"
+            )
+
+        if payload.celery_task_id is None:
+            logger.info(
+                f"connector_permission_sync_generator_task - Waiting for fence: "
+                f"fence={redis_connector.permissions.fence_key}"
+            )
+            sleep(1)
```
Comment (pablonyx) [inline backend/onyx/background/celery/tasks/doc_permission_syncing/tasks.py:253]: Might be worth avoid busy-wait loop with a retry wrapper here (though seems like benefit here would be pretty marginal

File: backend/onyx/background/celery/tasks/indexing/tasks.py (hunk 2 above)
Comment (LostVector) [reply backend/onyx/background/celery/tasks/indexing/tasks.py]: Yes!  Fixed

File: backend/onyx/background/celery/tasks/doc_permission_syncing/tasks.py (hunk 5 above)
Comment (rkuo-danswer) [reply backend/onyx/background/celery/tasks/doc_permission_syncing/tasks.py:253]: Will leave this as is for now.

**Pull request merged**